src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -601,6 +602,20 @@ public TypeSymbolWithAnnotations SetUnknownNullabilityForReferenceTypesIfNecessa
                 this.SetUnknownNullabilityForReferenceTypes();
         }
 
+        // PROTOTYPE(NullableReferenceTypes): external annotations should be removed or fully designed/productized
+        public TypeSymbolWithAnnotations SetUnknownNullabilityForReferenceTypesIfNecessary(ModuleSymbol module, ImmutableArray<bool> extraAnnotations)
```

Comment:
>SetUnknownNullabilityForReferenceTypesIfNecessary [](start = 41, length = 49)

I think the name of this method is confusing. Would "SetNullabilityForReferenceTypesOrResetToUnknownIfNecessary" better reflect what this function is doing?

Reply:
Also, it looks like there is only one call site for this method and it is in PEParameterSymbol. Consider moving this helper there.

---
In reply to: [183124205](https://github.com/dotnet/roslyn/pull/25928#discussion_r183124205) [](ancestors = 183124205)